evalml/pipelines/components/estimators/regressors/lightgbm_regressor.py

```diff
@@ -0,0 +1,96 @@
+import copy
+
+import pandas as pd
+from sklearn.preprocessing import OrdinalEncoder
+from skopt.space import Integer, Real
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import SEED_BOUNDS, get_random_seed, import_or_raise
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    _rename_column_names_to_numeric
+)
+
+
+class LightGBMRegressor(Estimator):
+    """LightGBM Regressor"""
+    name = "LightGBM Regressor"
+    hyperparameter_ranges = {
+        "learning_rate": Real(0.000001, 1),
+        "boosting_type": ["gbdt", "dart", "goss", "rf"],
+        "n_estimators": Integer(10, 100),
+        "max_depth": Integer(0, 10),
```

Comment:
@bchen1116 do you have any sense of what the fit time of lightgbm is like when max_depth or n_estimators is near the max value? We've observed larger values for similar parameters for other tree-based models can cause long fit times with no benefit in accuracy. I do see that these ranges align with some of our other tree-based estimators, including [catboost](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/estimators/regressors/catboost_regressor.py#L26) which we've had problems with in the past, so that's good.

Reply:
I am not sure. I can run some perf tests monday to experiment with these though!

Reply:
Perf tests are in the updated doc, and parameter defaults have been changed!